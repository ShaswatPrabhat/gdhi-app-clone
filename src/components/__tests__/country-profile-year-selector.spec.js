import { mount } from "@vue/test-utils";
import flushPromises from "flush-promises";
import { describe, expect, it, beforeEach, vi, afterEach } from "vitest";
import CountryProfileYearSelector from "../countryProfile/country-profile-year-selector.vue";
import { i18n } from "../../plugins/i18n";
import { createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";
import axios from "axios";

describe("Country Profile Year Selector ", () => {
  let wrapper;
  const setDefaultYearSpy = vi.fn();
  const getDefaultYearSpy = vi.fn();
  const axiosGetSpy = vi.spyOn(axios, "get");

  window.appProperties = {
    setDefaultYear: setDefaultYearSpy,
    getDefaultYear: getDefaultYearSpy,
  };

  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();

  beforeEach(async () => {
    axiosGetSpy.mockResolvedValueOnce({
      data: { years: ["Version 1", "2022", "2023"], defaultYear: "2022" },
    });
    axiosGetSpy.mockResolvedValueOnce({
      data: ["2023", "2021", "Version1"],
    });
    wrapper = mount(CountryProfileYearSelector, {
      localVue,
      router,
      i18n,
    });
    await flushPromises();
  });

  afterEach(() => {
    axiosGetSpy.mockReset();
  });

  it("should render select year description", async () => {
    expect(wrapper.find(".select-year-desc").find("p").text()).equal(
      i18n.messages.en.countryProfile.selectYearDescription
    );
  });

  it("should render select year", async () => {
    expect(
      wrapper.findAll(".title .sub-header-country-profile").at(0).text()
    ).equal(i18n.messages.en.countryProfile.selectYear);
  });

  it("should render Year Selector", async () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
  });

  it("should return the published years of a country when api call is made", async () => {
    let data = ["2023", "2021", "Version1"];
    expect(wrapper.vm.years).to.deep.equal(data);
  });
});
