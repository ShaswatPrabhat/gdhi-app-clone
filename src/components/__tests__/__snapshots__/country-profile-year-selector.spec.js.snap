// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Country Profile Year Selector  > should render Year Selector 1`] = `
<div
  class="year-select-container"
>
  <div
    class="title"
  >
    <div
      class="sub-header-country-profile"
    >
       Select Year 
    </div>
    <div
      class="select-year-desc"
    >
      <p>
        To view data for a specific year, please select the desired year
      </p>
    </div>
  </div>
  <div
    class="year-indicator year-indicator-select"
  >
    <div>
      <select
        class="year-indicator-select ltr width"
        name="test_select3"
      >
        <option
          value="2023"
        >
           2023 
        </option>
        <option
          value="2021"
        >
           2021 
        </option>
        <option
          value="Version1"
        >
           Version1 
        </option>
      </select>
        
    </div>
  </div>
</div>
`;
